core/src/main/java/com/google/zxing/client/result/ResultParser.java

```diff
@@ -74,27 +47,50 @@ public abstract class ResultParser {
    * of information (email, URL, etc.) and return a {@link ParsedResult} encapsulating
    * the result of parsing.
    *
-   * @param theResult the raw {@link Result} to parse
+   * @param barcodeResult the raw {@link Result} to parse
    * @return {@link ParsedResult} encapsulating the parsing result
    */
-  public abstract ParsedResult parse(Result theResult);
+  public abstract ParsedResult parse(Result barcodeResult);
 
-  protected static String getMassagedText(Result result) {
-    String text = result.getText();
+  protected static String getMassagedText(Result barcodeResult) {
+    String text = barcodeResult.getText();
     if (text.startsWith(BYTE_ORDER_MARK)) {
       text = text.substring(1);
     }
     return text;
   }
 
-  public static ParsedResult parseResult(Result theResult) {
-    for (ResultParser parser : PARSERS) {
-      ParsedResult result = parser.parse(theResult);
+  public static ParsedResult parseBarcodeResult(Result barcodeResult) {
+    ResultParser[] parsers = {
```

Comment:
This re-declares the array every time the method is called, and reinstantiates all the objects. It isn't necessary, which is why the code is the way it is now.